server/index.js

```diff
@@ -3,28 +3,70 @@ const bodyParser = require('body-parser');
 const path = require('path');
 
 const app = express();
-const db = require('../database/controllers');
-
-console.log(db);
+const {
+  getDoc, saveDoc, updateDoc, deleteDoc,
+} = require('../database/controllers');
 
 const PORT = 3002;
 
 app.use(bodyParser.json());
 app.use(bodyParser.urlencoded({ extended: true }));
 app.use(express.static(path.join(__dirname, '/../public')));
 
+app.post('/items/', (req, res) => {
+  saveDoc(req.body, (err) => {
+    if (err) {
+      console.log(`error saving: ${err}`);
+      res.status(400);
+      res.end();
+    } else {
+      console.log('saved id: ', req.body.id);
+      res.status(201);
+      res.end();
+    }
+  });
+});
+
 app.get('/items/:id', (req, res) => {
   const itemId = req.params.id;
-  db(itemId, (err, success) => {
+  getDoc(itemId, (err, success) => {
     if (err) {
-      console.log(err);
+      console.log(`error fetching: ${err}`);
       res.sendStatus(404).end();
     } else {
-      res.send(success[0]).end();
+      res.send(success).end();
+    }
+  });
+});
+
+app.put('/items/:id', (req, res) => {
+  const key = Object.keys(req.body);
+  const value = Object.values(req.body);
+  updateDoc(req.params.id, key[0], value[0], (err, success) => {
```

Comment:
Cool! I haven't considered doing an update this way with using Object.keys and Object.values.